src/migration.rs

```diff
@@ -66,11 +88,63 @@ pub fn migrate(from: &Path, mut to: Options) -> Result<()> {
 			}
 			true
 		})?;
+		if !keep_dest {
+			dest.commit_raw(commit)?;
+			commit = Vec::with_capacity(COMMIT_SIZE);
+			std::mem::drop(dest);
+			dest = Db::open_or_create(&to)?;
+			while !dest.all_empty_logs() {
+				std::thread::sleep(std::time::Duration::from_millis(300));
+			}
+			log::info!("Collection migrated {}, imported", c);
+
+			std::mem::drop(dest);
+			std::mem::drop(source);
+			move_collection(c, &to.path, from)?;
```

Comment:
MIgration should not touch source. Everything should be copied for case of failures we don't want to end up with both the source and dest ruined. During the upgrade only after we have migrated the whole database successfully we can switch to it and delete the old one.

Reply:
My use case is changing column option that are not strict (eg compression or the size threshold) for a column.
So in this case I am not really 'migrating' but simply rewriting a collection ('keep_dest' set to false or 'overwrite' parameter to true).
When the metadata version number is increased, this is not possible (there is a check above).